x-pack/plugins/upgrade_assistant/__jest__/client_integration/cluster.test.ts

```diff
@@ -0,0 +1,363 @@
+        });
+
+        component.update();
+
+        const request = server.requests[server.requests.length - 1];
+        const mlDeprecation = upgradeStatusMockResponse.cluster[0];
+
+        expect(request.method).toBe('DELETE');
+        expect(request.url).toBe(
+          `/api/upgrade_assistant/ml_snapshots/${
+            (mlDeprecation.correctiveAction! as MlAction).jobId
+          }/${(mlDeprecation.correctiveAction! as MlAction).snapshotId}`
+        );
+        expect(request.status).toEqual(500);
+
+        const accordionTestSubj = `depgroup_${upgradeStatusMockResponse.cluster[0].message
+          .split(' ')
+          .join('_')}`;
+
+        expect(find(`${accordionTestSubj}.fixMlSnapshotsButton`).text()).toEqual('Failed');
+      });
+    });
+  });
+
+  describe('no deprecations', () => {
+    beforeEach(async () => {
+      const noDeprecationsResponse = {
+        readyForUpgrade: false,
+        cluster: [],
+        indices: [],
+      };
+
+      httpRequestsMockHelpers.setLoadEsDeprecationsResponse(noDeprecationsResponse);
+
+      await act(async () => {
+        testBed = await setupClusterPage({ isReadOnlyMode: false });
+      });
+
+      const { component } = testBed;
+
+      component.update();
+    });
+
+    test('renders prompt', () => {
+      const { exists, find } = testBed;
+      expect(exists('noDeprecationsPrompt')).toBe(true);
+      expect(find('noDeprecationsPrompt').text()).toContain('Ready to upgrade!');
+    });
+  });
+
+  describe('error handling', () => {
+    test('handles 403', async () => {
+      const error = {
+        statusCode: 403,
+        error: 'Forbidden',
+        message: 'Forbidden',
+      };
+
+      httpRequestsMockHelpers.setLoadEsDeprecationsResponse(undefined, error);
+
+      await act(async () => {
+        testBed = await setupClusterPage({ isReadOnlyMode: false });
+      });
+
+      const { component, exists, find } = testBed;
+
+      component.update();
+
+      expect(exists('permissionsError')).toBe(true);
+      expect(find('permissionsError').text()).toContain(
+        'You are not authorized to view Elasticsearch deprecations.'
+      );
+    });
+
+    test('handles upgrade error', async () => {
+      const error = {
+        statusCode: 426,
+        error: 'Upgrade required',
+        message: 'There are some nodes running a different version of Elasticsearch',
+        attributes: {
+          allNodesUpgraded: true,
+        },
+      };
+
+      httpRequestsMockHelpers.setLoadEsDeprecationsResponse(undefined, error);
+
+      await act(async () => {
+        testBed = await setupClusterPage({ isReadOnlyMode: false });
+      });
+
+      const { component, exists, find } = testBed;
+
+      component.update();
+
+      expect(exists('upgradedCallout')).toBe(true);
+      expect(find('upgradedCallout').text()).toContain(
+        'Your configuration is up to date. Kibana and all Elasticsearch nodes are running the same version.'
```

Comment:
I don't really understand this test from just reading it. The title says that there is an upgrade error and the mocked response is that an upgrade is required. But the callout seems to be showing a message that the configuration is up to date.

Reply:
Thanks for calling this out! It is indeed confusing and I had to relook at the code myself 😅.

I renamed the test case and added a comment. This was pre-existing behavior, but basically we return a `426` if there is ever a version mismatch between Kibana and any running nodes. If all nodes don't match with Kibana, we assume they have been upgraded.

I see at some point we changed the copy from "All Elasticsearch nodes have been upgraded. You may now upgrade Kibana." to what you see in the test case. I see now that is inaccurate. I will update the copy in a separate PR since it wasn't directly added as part of this PR.